runtime/snapshot.rs

```diff
@@ -0,0 +1,246 @@
+    _path: &Path,
+    _display: &str,
+    _api_name: &str,
+  ) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+
+  fn check_write(
+    &mut self,
+    _path: &Path,
+    _api_name: &str,
+  ) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+
+  fn check_write_partial(
+    &mut self,
+    _path: &Path,
+    _api_name: &str,
+  ) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+
+  fn check_write_all(&mut self, _api_name: &str) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+
+  fn check_write_blind(
+    &mut self,
+    _path: &Path,
+    _display: &str,
+    _api_name: &str,
+  ) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+}
+
+impl deno_kv::sqlite::SqliteDbHandlerPermissions for Permissions {
+  fn check_read(
+    &mut self,
+    _path: &Path,
+    _api_name: &str,
+  ) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+
+  fn check_write(
+    &mut self,
+    _path: &Path,
+    _api_name: &str,
+  ) -> Result<(), AnyError> {
+    unreachable!("snapshotting!")
+  }
+}
+
+pub fn create_runtime_snapshot(snapshot_path: PathBuf) {
+  // NOTE(bartlomieju): ordering is important here, keep it in sync with
+  // `runtime/worker.rs`, `runtime/web_worker.rs` and `cli/build.rs`!
+  let fs = std::sync::Arc::new(deno_fs::RealFs);
+  let mut extensions: Vec<Extension> = vec![
+    deno_webidl::deno_webidl::init_ops_and_esm(),
+    deno_console::deno_console::init_ops_and_esm(),
+    deno_url::deno_url::init_ops_and_esm(),
+    deno_web::deno_web::init_ops_and_esm::<Permissions>(
+      Default::default(),
+      Default::default(),
+    ),
+    deno_fetch::deno_fetch::init_ops_and_esm::<Permissions>(Default::default()),
+    deno_cache::deno_cache::init_ops_and_esm::<SqliteBackedCache>(None),
+    deno_websocket::deno_websocket::init_ops_and_esm::<Permissions>(
+      "".to_owned(),
+      None,
+      None,
+    ),
+    deno_webstorage::deno_webstorage::init_ops_and_esm(None),
+    deno_crypto::deno_crypto::init_ops_and_esm(None),
+    deno_broadcast_channel::deno_broadcast_channel::init_ops_and_esm(
+      deno_broadcast_channel::InMemoryBroadcastChannel::default(),
+    ),
+    deno_ffi::deno_ffi::init_ops_and_esm::<Permissions>(),
+    deno_net::deno_net::init_ops_and_esm::<Permissions>(None, None),
+    deno_tls::deno_tls::init_ops_and_esm(),
+    deno_kv::deno_kv::init_ops_and_esm(deno_kv::sqlite::SqliteDbHandler::<
+      Permissions,
+    >::new(None, None)),
+    deno_cron::deno_cron::init_ops_and_esm(
+      deno_cron::local::LocalCronHandler::new(),
+    ),
+    deno_napi::deno_napi::init_ops_and_esm::<Permissions>(),
+    deno_http::deno_http::init_ops_and_esm::<DefaultHttpPropertyExtractor>(),
+    deno_io::deno_io::init_ops_and_esm(Default::default()),
+    deno_fs::deno_fs::init_ops_and_esm::<Permissions>(fs.clone()),
+    deno_node::deno_node::init_ops_and_esm::<Permissions>(None, fs),
+    runtime::init_ops_and_esm(),
```

Comment:
So this bit is actually only specifying the JS sources for the extensions but not actual ops - my idea was that this module would properly set up both JS and ops (and thus couldn't be used in `runtime/build.rs` for the time being, until we migrate to only using `cli/build.rs`).